user can add listing

damwhit opened the pull request:
- user is able to create a listing, All pages used in test are dynamic and I plan to drive out the listing index and show pages during the user can edit a listing user story
- added will_paginate gem and now the sell page/ events index is paginated

closes #7

damwhit committed:
[wip] user can add listing

still having trouble adding a listing, all of the form identifiers are unique but only the last one is getting sent through the params and it is labeled seat_number

damwhit committed:
fixed merge conflicts

damwhit committed:
user can now create a listing

damwhit committed:
user can add a listing

all tests passing, need to still figure out validation so that seats must be next to one another.  Also listings index and show pages need to become dynamic

damwhit committed:
user can create a listing

user is able to create a listing, All pages used in test are dynamic and I plan to drive out the listing index and show pages during the user can edit a listing user story

damwhit commented on spec/features/user/user_can_add_a_listing_spec.rb:
```diff
@@ -0,0 +1,73 @@
+    click_on "create listing"
+
+    expect(current_path).to eq("/dashboard")
+
```
It reads better this way, in my opinion

damwhit commented on app/controllers/listings_controller.rb:
```diff
@@ -1,10 +1,33 @@
+      params.permit(:price, :row)
+    end
+
+    def seat_params
```
This was done to denote that these are private methods.

damwhit commented on app/controllers/listings_controller.rb:
```diff
@@ -1,10 +1,41 @@
+
+  private
+
+    def ticket_params
```
This is to denote private methods.  ignoring hound here

damwhit committed:
Sell page now paginates

added will_paginate gem and now the sell page/ events index is paginated

damwhit commented on app/controllers/listings_controller.rb:
```diff
@@ -1,10 +1,42 @@
+      flash[:info] = "Your tickets are up for sale!"
+      redirect_to user_dashboard_path
+    else
+      flash.now[:alert] = "Sorry, boss lolololololololol.  Something went wrong ;>(... Please try again."
```
Going to leave alone - seems unnecessary to concatenate this

damwhit commented on app/assets/javascripts/user/add_seat.js:
```diff
@@ -0,0 +1,8 @@
+
+$( document ).ready(function() {
+  $('#add-seat').click(function() {
+    $('#seat_number_' + seats).after("<br><input type='text' name='seat["+ seats +"]' id='seat_number_" + (seats+1) + "' required='required' placeholder='seat number " + (seats+1) + " ' class='form-control text-input'>");
```
gonna leave this one alone

erinnachen commented on app/controllers/events_controller.rb:
```diff
@@ -1,6 +1,7 @@
 
   def index
+    @events = Event.all.paginate(page: params[:page])
   end
```
I think this will show events in the past...

damwhit replied:
Okay, that's one thing that I wanted to bring up tomorrow.  Right now the events index is the sell page and that's why it's all of the events.

erinnachen replied:
I think we can just scope by status: "upcoming" and date in the future. Admin is able to see all events, but user won't need to.

erinnachen commented on spec/features/user/user_can_add_a_listing_spec.rb:
```diff
@@ -0,0 +1,65 @@
+
+    within(".table-listings") do
+      expect(page).to have_content("$30.00")
+    end
```
Are there other things we should see about this listing.. event? status?

erinnachen commented on app/views/users/show.html.erb:
```diff
@@ -51,13 +51,13 @@
+              <td><h4><%= listing.event.status %></h4></td>
+              <td><h4><%= listing.tickets.formatted_sum %></h4></td>
+              <td><h4><%= link_to "details", listing_path(listing) %></h4></td>
             </tr>
```
we should pull this out into a partial

damwhit replied:
Agreed.  done

erinnachen commented on app/views/shared/_user_nav.html.erb:
```diff
@@ -20,6 +20,9 @@
     </li>
+    <li class="pull-right">
+      <a id="sell" href="/sell" style="color:#777; margin-top: 10px;">sell tickets</a>
+    </li>
```
link_to here?
also the styling should be moved into a stylesheet.

damwhit replied:
I like it.  Done!

erinnachen commented on app/controllers/listings_controller.rb:
```diff
@@ -1,10 +1,42 @@
+    def seat_params
+      params.require(:seat).permit("0", "1", "2", "3",
+                                   "4", "5", "7", "8", "9")
+    end
```
I think there is a more dynamic way to do this.

damwhit replied:
Agreed.. Definitely needs some refactoring later.  Couldn't find another way of doing it without it being super ugly

damwhit committed:
dates must now be upcoming for events index

when a user wishes to create a listing, they are only able to see future events on the sell page

damwhit commented on app/controllers/events_controller.rb:
```diff
@@ -1,6 +1,7 @@
 class EventsController < ApplicationController
 
   def index
+    @events = Event.where("time >= ?", Time.zone.now.beginning_of_day).paginate(page: params[:page]).order(time: :asc)
```
going to ignore this.  Not sure how to make it shorter.

damwhit committed:
SQL is no longer in controller for upcoming events

moved upcoming events to be a scope

damwhit commented on app/models/event.rb:
```diff
@@ -11,6 +11,8 @@ class Event < ActiveRecord::Base
   # validates :description, presence: true
   # validates :category_id, presence: true
 
+  scope :upcoming_events, -> { where("time >= ?", Time.zone.now.beginning_of_day) }
```
Already tried seperating into two lines but hound didn't like the scope being in two lines

erinnachen commented:
Coolio. This was a behemoth!
